### PR TITLE
980 register sample page

### PR DESCRIPTION
**Pull request by ciaranschutte**
Code ported from `platform-ui`
additions are to fix legacy/no typing & new Table from uikit

Sorry for the large PR. Went about this ticket the wrong way - new parts to it kept being discovered - code additions kept expanding. Luckily it's for the most part the same structure we have in `platform-ui` already. I've updated some types, structure of the code and some tweaking of library usage was needed but large sections are left unchanged as is the scope of the ticket. 

Lots of eslint / ts errors that I'm working on atm (that's why build is failing).
The original codebase didn't have the same code quality strictness we have in this project so this is a new issue. Unable to just copy types.

### CODE REVIEW

File: src/app/(post-login)/submission/program/[shortName]/registration/components/FilePreview.tsx [hunk 1]
```diff
@@ -77,7 +80,13 @@ const FilePreview = ({
     <FileTable
       records={records}
       stats={stats}
-      submissionInfo={submissionInfo}
+      submissionInfo={
```
Comment (ciaranschutte) [inline src/app/(post-login)/submission/program/[shortName]/registration/components/FilePreview.tsx:83]: gql generated typings introduces a bunch of "Maybe" types. I haven't found a good way around this apart from type assertions

File: src/app/(post-login)/submission/components/ProgramMenu.tsx [hunk 2]
```diff
@@ -16,63 +16,99 @@
...
+      `}
+    >
+      Register Samples
+      {/* {isSubmissionSystemDisabled ? (
+        <Icon name="lock" fill="accent3_dark" width="15px" />
+      ) : clinicalRegistrationHasError ? (
+        <Icon name="exclamation" fill="error" width="15px" />
+      ) : clinicalRegistrationInProgress ? (
+        <Icon name="ellipses" fill="warning" width="15px" />
+      ) : null} */}
```
Comment (demariadaniel) [inline src/app/(post-login)/submission/components/ProgramMenu.tsx:52]: Is this block related to something in progress?

Comment (ciaranschutte) [reply src/app/(post-login)/submission/components/ProgramMenu.tsx:52]: It's related to a piece of work that I could easily scope out of this ticket as it's already a very large ticket with lots of code change.

**Review by demariadaniel — APPROVED** [review]
